moztelemetry/store.py

```diff
@@ -36,7 +38,12 @@ def get_key(self, key):
         try:
             # get_key must return a file-like object because that's what's
             # required by parse_heka_message
-            return bucket.Object(key).get()['Body']
+            s3object = bucket.Object(key).get()
+            if s3object['ResponseMetadata']['HTTPHeaders'].get(
```

Comment:
I've updated the code to use the top-level key for content-encoding instead of the low-level one.

Like with telemetry-batch-view, the testing harness here creates an implementation of the `S3Store` (`InMemoryStore`) for testing that makes adding a test that actually uses the code above difficult. I can add a test like `test_get_gzip_key` that adds the streaming wrapper within the testing logic and confirms the data written to disk is decompressed, but that's essentially a subset of what the test in telemetry-tools does and to me feels like confirming that python can gzip and gunzip a file. It would also potentially involve changing the `S3Store` API to support specifying a `content-encoding` when uploading an object, which is something I didn't want to do. An option to avoid changing the API would be to use e.g. file extension to determine content-encoding for the `InMemoryStore`, but that reverts to the same  gzip-gunzip cycle.